taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -467,17 +473,42 @@ class KernelGen : public IRVisitor {
       emit("{} {} = {}({});", dt_name, stmt->short_name(),
            opengl_data_type_name(stmt->cast_type), stmt->operand->short_name());
     } else if (stmt->op_type == UnaryOpType::cast_bits) {
-      if (stmt->cast_type->is_primitive(PrimitiveTypeID::f32) &&
-          stmt->operand->element_type()->is_primitive(PrimitiveTypeID::i32)) {
+      auto dst_type = stmt->cast_type;
+      auto src_type = stmt->operand->element_type();
+      auto dst_int = 0;
+      if (is_integral(dst_type))
+        dst_int = is_unsigned(dst_type) ? 2 : 1;
+      auto src_int = 0;
+      if (is_integral(src_type))
+        src_int = is_unsigned(src_type) ? 2 : 1;
```

Comment:
Instead of using magic numbers `0, 1, 2`, maybe use a few local `constexpr` constant identifiers for readability? E.g. `constexpr int signed_integer_type = 0;`.

Variable names `dst_int` and `src_int` are a bit misleading since they can represent float-point types as well.